### PR TITLE
Update web link URLs in documentation

### PR DESCRIPTION
**Pull request by g-re-g**
Updates URLs in documentation to their current location. Stumbled across a couple of these as I was reading through the source and figured I'd update them.

The following applies to the URLs in this PR. I'm sure there are many I didn't see and I didn't touch URLs in comments that didn't look like they were being used as documentation.
 * Any urls that auto redirect from `http` to `https` have been changed to `https`.
 * Dead links have been changed to an up to date location.
 * Urls have been checked for validity

Happy to add / remove / or change anything to how folks see fit! Happy to close this PR if it's not needed/wanted!

### CODE REVIEW

File: lib/uri.rb [hunk 1]
```diff
@@ -44,17 +44,17 @@
...
-# - RFC2732[http://tools.ietf.org/html/rfc2732]
-# - RFC3986[http://tools.ietf.org/html/rfc3986]
+# - RFC822[https://www.rfc-editor.org/rfc/rfc822]
+#  RFC1738[https://www.rfc-editor.org/rfc/rfc1738]
+#  RFC2255[https://www.rfc-editor.org/rfc/rfc2255]
+#  RFC2368[https://www.rfc-editor.org/rfc/rfc2368]
+#  RFC2373[https://www.rfc-editor.org/rfc/rfc2373]
+#  RFC2396[https://www.rfc-editor.org/rfc/rfc2396]
+#  RFC2732[https://www.rfc-editor.org/rfc/rfc2732]
+#  RFC3986[https://www.rfc-editor.org/rfc/rfc3986]
```
Comment (nobu) [inline lib/uri.rb:57]: Why changed this unordered list to verbatim?

```suggestion
# - RFC1738[https://www.rfc-editor.org/rfc/rfc1738]
# - RFC2255[https://www.rfc-editor.org/rfc/rfc2255]
# - RFC2368[https://www.rfc-editor.org/rfc/rfc2368]
# - RFC2373[https://www.rfc-editor.org/rfc/rfc2373]
# - RFC2396[https://www.rfc-editor.org/rfc/rfc2396]
# - RFC2732[https://www.rfc-editor.org/rfc/rfc2732]
# - RFC3986[https://www.rfc-editor.org/rfc/rfc3986]
```

**Review by nobu — CHANGES_REQUESTED** [review]
These files have upstreams.

File: lib/uri.rb (hunk 1 above)
Comment (g-re-g) [reply lib/uri.rb:57]: Whoops, definitely not my intention, good lookin out!

**Comment by g-re-g** [comment]
> These files have upstreams.

I'm not exactly sure what that means. I think it means that they're maintained in some other repo then brought into here in some process and that editing them here will just get overwritten if they are brought in again. Or something like that. Is that the case? Anyway, happy to close this PR if it's not gonna work!

**Comment by hsbt** [comment]
Can you submit them each repository?

* https://github.com/ruby/drb
* https://github.com/ruby/net-http/
* https://github.com/ruby/rdoc/
* https://github.com/ruby/shellwords
* https://github.com/ruby/uri
* https://github.com/ruby/time

**Pull request closed without merging**